wolfcrypt/src/chacha.c

```diff
@@ -418,14 +418,16 @@ int wc_Chacha_Process(ChaCha* ctx, byte* output, const byte* input,
 
     #ifdef HAVE_INTEL_AVX2
     if (IS_INTEL_AVX2(cpuidFlags)) {
-        SAVE_VECTOR_REGISTERS();
+        if (SAVE_VECTOR_REGISTERS() != 0)
```

Comment:
Does `SAVE_VECTOR_REGISTERS` return `BAD_STATE_E`? I see that `return BAD_STATE_E` flow throughout. In some places might be cleaner to have `SAVE_VECTOR_REGISTERS` return `BAD_STATE_E`. Did you consider adding a new return code for the `SAVE_VECTOR_REGISTERS` failure? The code could only exist for LKM.

Reply:
I've refactored to pass up the retval from `SAVE_VECTOR_REGISTERS()`, and made its retvals consistent.  I missed your idea of a new error code until just now -- we're almost out of wolfcrypt-range codes alas!  Can revisit.